analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -126,19 +125,42 @@ private static ProgramState[] ProcessLinqEnumerableAndQueryable(SymbolicContext
         }
     }
 
-    private static ProgramState[] ProcessStringIsNullOrEmpty(SymbolicContext context, IInvocationOperationWrapper invocation) =>
-        context.State[invocation.Arguments[0].ToArgument().Value]?.Constraint<ObjectConstraint>() switch
+    private static ProgramState[] ProcessIsNotNullWhen(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)))
+        {
+            return ProcessIsNotNullWhen(state, invocation.WrappedOperation, invocation.Arguments[0].ToArgument(), false);
+        }
+        foreach (var argument in invocation.Arguments.Select(x => x.ToArgument())) // TODO: support attributes on more than one argument
```

Comment:
That should be easy, shouldn't it? Just do `state = ProcessIsNotNullWhen(state, ...)` instead of return

Reply:
...aha, except it explodes as an array. But still doable with two nested foreach. Similar to `RoslynSymbolicExecution.ProcessOperation`

Reply:
I think two nested loops would still generate wrong states:
* Operation: true (Symbol arg1)
* Operation: false (Symbol arg1)
* Operation: true (Symbol arg2)
* Operation: false (Symbol arg2)
instead of
* Operation: true (Symbol arg1. Symbol2 arg)
* Operation: false (Symbol arg1. Symbol2 arg)

 We should come back to this later.

Reply:
I'll give it a try in the next PR #6144